src/charm.py

```diff
@@ -509,7 +512,10 @@ def _on_config_changed(self, _event: ConfigChangedEvent) -> None:
                 self._stored.path, self.config["group"]
             )
             self._stored.path = self.config["path"]
-        if prev_config_for_flush:
+        if self.config[LABELS_CONFIG_NAME] != self._stored.labels:
+            should_flush_runners = True
+            self._stored.labels = self.config[LABELS_CONFIG_NAME]
+        if prev_config_for_flush or should_flush_runners:
```

Comment:
```suggestion
        if state.charm_config.labels != self._stored.labels:
            should_flush_runners = True
            self._stored.labels = state.charm_config.labels
        if prev_config_for_flush or should_flush_runners:
```

Reply:
I think this should be stored as the config string as is because `state.charm_config.labels` returns a tuple of strings. The reason is because I need to initialized the stored state during charm init with default values before I initialize the charm state.